src/mococrw/mac.h

```diff
@@ -134,13 +139,88 @@ class HMAC : public MessageAuthenticationCode {
 
 private:
     /**
-     * Internal class for applying the PIMPL design pattern
-     * (to hide the details of storing the padding objects from the client)
+     * @brief Internal class for applying the PIMPL design pattern
+     */
+    class Impl;
+
+    /**
+     * @brief Pointer for PIMPL design pattern
+     */
+    std::unique_ptr<Impl> _impl;
+};
+
+class CMAC : public MessageAuthenticationCode {
+public:
+    /**
+     * @brief Constructor
+     * @param cipherType the type of encryption function which shall be used
+     * @param key the key used for CMAC
+     */
```

Comment:
In the current implementation, this throws if cipherType is NONE. I wonder if we should really add this NONE type. This way, we could syntactically ensure that this call will never fail due to an invalid cipher.

The documentation should also reflect that it throws a MoCOCrWException if the key size is wrong.

Reply:
I just had another look into the code including OpenSSL itself. In fact `openssl::DigestTypes::NONE` is used e.g. for `EdDSASignaturePublicKeyCtx::verifyMessage()` to allow OpenSLL to select a default digest by its own.
Regardless that `NONE` is quite a bad name for his behaviour, there seems to be a valid use case for this.

Such a behaviour does not exist for AES-CMAC (in current OpenSSL implementation at least).
I guess you still vote for removing `openssl::CmacCipherTypes::None`, right?

Reply:
Yes. Let's try to (slowly but constantly) move towards a better code-base.

Reply:
Exception with message is thrown. Documentation and test case updated.